creusot-contracts/src/lib.rs

```diff
@@ -22,6 +22,25 @@
 //! ```
 //!
 //! For a more detailed explanation, see the [guide](https://creusot-rs.github.io/creusot/guide).
+//!
+//! # Module organization
+//!
+//! 1. Core features of Creusot
+//!
+//!     - [`invariant`][mod@invariant]: Type invariants
+//!     - [`macros`]: `#[requires]`, `#[ensures]`, etc.
+//!     - [`resolve`][mod@resolve]: Resolve mutable borrows
+//!     - [`snapshot`][mod@snapshot]: Snapshots
+//!
+//! 2. [`logic`][mod@logic]: Logical structures used in specifications
+//!
+//! 3. [`ghost`][mod@ghost]: Ghost code
+//!
+//! 4. [`std`][mod@std]: Specifications for the `std` crate
+//!
+//! 5. [`cell`][mod@cell]: Interior mutability
+//!
+//! 6. [`prelude`][mod@prelude]: What you get from `use creusot_contracts::*;`
```

Comment:
While we are at it, I wonder whether `prelude` and `base_prelude` are still relevant or should be removed. The goal was to find some way to import it by default at the header of every crate loaded by Creusot, but I don't think we will be able to do that.

Reply:
Having all the reexports in a `prelude` module makes the generated doc a bit tidier.

The doc page currently just shows "Re-exported: `pub use prelude::*`". If we inlined `prelude`, we would suffer all of the traits re-exported `as _`.

Reply:
I'd still like the list of re-exports to appear somewhere.